src/BizHawk.Emulation.Cores/Computers/Doom/LibDSDA.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Runtime.InteropServices;
+
+using BizHawk.BizInvoke;
+
+namespace BizHawk.Emulation.Cores.Computers.Doom
+{
+	public abstract class CInterface
+	{
+		[UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+		public delegate int load_archive_cb(string filename, IntPtr buffer, int maxsize);
+
+		[StructLayout(LayoutKind.Sequential)]
+		public class InitSettings

```

Comment:
Can these be `struct`s?

Reply:
Not sure I understand

Reply:
I would hope you're not relying on reference identity for these, so why declare them as `class`es?

Reply:
I simply imitated what other cores did. If I change any of these into a struct, I get a runtime error: 

System.NotImplementedException: 'Unrecognized parameter type!'